Src/FluentAssertions/Execution/ChainedAssertionScope.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+
+namespace FluentAssertions.Execution
+{
+    /// <summary>
+    /// Allows chaining multiple assertion scopes together using <see cref="Continuation.Then"/>.
+    /// </summary>
+    /// <remarks>
+    /// If the parent scope has captured a failed assertion, this class ensures that successive assertions
+    /// are no longer evaluated.
+    /// </remarks>
+    public class ContinuedAssertionScope : IAssertionScope
+    {
+        private readonly AssertionScope predecessor;
```

Comment:
❓ Could the `predecessor` be another `ContinuedAssertionScope`?

Reply:
Yes, but those would all forward the calls to the underlying `AssertionScope`.